Maestro/src/Microsoft.DotNet.Maestro.WebApi/Controllers/CommitPushedController.cs

```diff
@@ -20,12 +24,22 @@ public class CommitPushedController : ApiController
         private EventRegistrationTable _eventService = new EventRegistrationTable();
         private AzureStorageService _storageService = new AzureStorageService();
 
-        public async Task Post(PushWebHookEvent e)
+        public async Task Post()
         {
+            // The PushWebHookEvent object must be manually deserialized instead of using parameter binding 
+            // because the payload signature verification requires access to the request's raw content.  When
+            // using parameter binding, the content is read during the binding and cannot be accessed again due
+            // to Web API treating the request's content as a forward-only stream which can only be read once.
+            byte[] requestContent = await Request.Content.ReadAsByteArrayAsync();
+            PushWebHookEvent e = JsonConvert.DeserializeObject<PushWebHookEvent>(
```

Comment:
We should probably only deserialize the body after we've verified that the request is OK.